File: CS106A/Programming Exercises from the Art of Java, Chapter 8 (Strings and Characters)/Capitalize.java
/*
 * File: Capitalize.java
 * Name: Oliver Pecha
 * Section Leader: Online Learning
 * -----------------
 * Book / Chapter 8 / Programming Exercise 3
 * -----------------
 *  Implement a method capitalize(str) that returns a string in which the initial character is capitalized (if it is a letter) and all other letters 
 *  are converted so that they appear in lowercase form. Characters other than letters are not affected. For example, capitalize("BOOLEAN") and 
 *  capitalize("boolean") should each return the string "Boolean".
 */

public class Capitalize {
	
	public Capitalize(String str) {
		word = str;
	}

	public void setCapitalize() {
		String str = toLowercases(word);
		word = firstCapital(str);
		return;
	}
	
	public String toLowercases(String str) {
		str = str.toLowerCase();
		return str;
	}
	
	private String firstCapital(String str) {
		String first = "" + str.charAt(0);
		first = first.toUpperCase();
		String remainder = str.substring(1); 
		return first + remainder;
	}
	
	public String toString() {
		return "" + word;
	}
	
	private String word = "";
}

File: CS106B/Chapter03 (Strings) Book Exercises/04. capitalize/capitalize.cpp
/*
 * File: capitalize.cpp
 * --------------
 * Implement a function capitalize(str) that returns a string in which the initial character is capitalized
 * (if it is a letter) and all other letters are converted to lowercase. Characters other than letters are not affected.
 * For example, both capitalize("BOOLEAN") and capitalize("boolean") should return the string "Boolean".
 */

#include "console.h"
#include "simpio.h"
using namespace std;


string capitalize(string str);

int main()
{
    string str = getLine("What is the phrase you want to convert to Capitalize Case");
    cout << "The Capitalized Case phrase is... " << capitalize(str) << endl;
    return 0;
}

string capitalize(string str) {
    int index = 0;
    for (int i = 0; i < str.length(); i++){
        if (index == 0) {
           str[i] = toupper(str[i]);
           index++;
        }
        else {
            str[i] = tolower(str[i]);
            index++;
        }
        if (isspace(str[i+1])) {
            index = -1;
        }
    }
    return str;
}
